gui/src/3D/controls/slider3D.ts

```diff
@@ -0,0 +1,193 @@
+        this.value = options?.value ? options.value : 50;       
+    }    
+
+    /** Gets or sets minimum value */
+    public get minimum(): number {
+        return this._minimum;
+    }
+
+    public set minimum(value: number) {
+        if (this._minimum === value) {
+            return;
+        }
+
+        this._minimum = Math.min(value, this._maximum);
+        this.value = this._value;
+    }
+
+     /** Gets or sets maximum value */
+     public get maximum(): number {
+         return this._maximum;
+     }
+ 
+     public set maximum(value: number) {
+         if (this._maximum === value) {
+             return;
+         }
+ 
+         this._maximum = Math.max(value, this._minimum);
+         this.value = this._value;
+     }
+
+    /** Gets or sets current value */
+    public get value(): number {
+        return this._value;
+    }
+
+    public set value(value: number) {
+        value = Math.max(Math.min(value, this._maximum), this._minimum);
+
+        if (this._value === value) {
+            return;
+        }
+
+        this._value = value;
+        this.onValueChangedObservable.notifyObservers(value);
+    }
+
+    protected get start(): number{
+        if(!this.node){
+            return -1;
+        }
+
+        return this._sliderBar.position.x - this._sliderBar.scaling.y / 2;
+    }
+
+    protected get end(): number{
+        if(!this.node){
+            return 1;
+        }
+
+        return this._sliderBar.position.x + this._sliderBar.scaling.y / 2;
+    }
+
+    // Mesh association
+    protected _createNode(scene: Scene): TransformNode {
+        const anchor = new AbstractMesh(`${this.name}_slider`, scene);    
+        
+        this._sliderBar = MeshBuilder.CreateCylinder(`${this.name}_sliderbar`, { diameter: 0.03, height: 1.0 }, scene);
+        this._sliderBar.rotation.z = -Math.PI / 2;
+        this._sliderBar.scaling.y = 2;
+        this._sliderBar.isPickable = false;
+        this._sliderBar.setParent(anchor);
+
+        this._sliderThumb = MeshBuilder.CreateBox(`${this.name}_sliderthumb`, { size: 0.1 }, scene);
+        this._sliderThumb.scaling = new Vector3(2, 2, 2);
+        this._sliderThumb.position.x = this._convertToPosition(this.value);
+        this._sliderThumb.addBehavior(this._createBehavior());
+        this._sliderThumb.setParent(anchor);
+
+        return anchor;
+    }
+
+    protected _affectMaterial(mesh: AbstractMesh) {
+        this._sliderBarMaterial = this._createBarMaterial(mesh.getScene());
+        this._sliderBar.material = this._sliderBarMaterial;
+
+        this._sliderThumbMaterial = this._createThumbMaterial(mesh.getScene());
+        this._sliderThumb.material =  this._sliderThumbMaterial;
+    }
+    
+    private _createBarMaterial(scene: Scene): Material {
+        const material  =  new StandardMaterial(`${this.name}_sliderbar_material`, scene);
+        material.specularColor = Color3.Black();
+
+        return material;       

```

Comment:
>        [](start = 24, length = 7)

nit: there are a lot of extra spaces throughout. please remove.

Reply:
Agreed, will do that. Thanks